contracts/LiquidationsPeripheral.vy

```diff
@@ -657,16 +703,24 @@ def addLiquidation(
     ILiquidationsCore(self.liquidationsCoreAddress).addLoanToLiquidated(_borrower, self.loansCoreAddresses[_erc20TokenContract], _loanId)
 
 
+@payable
 @external
 def payLoanLiquidationsGracePeriod(_loanId: uint256, _erc20TokenContract: address):
-    loan: Loan = ILoansCore(self.loansCoreAddresses[_erc20TokenContract]).getLoan(msg.sender, _loanId)
+    receivedAmount: uint256 = msg.value
+    ethPayment: bool = receivedAmount > 0
 
+    loan: Loan = ILoansCore(self.loansCoreAddresses[_erc20TokenContract]).getLoan(msg.sender, _loanId)
     assert loan.defaulted, "loan is not defaulted"
 
+    if ethPayment:
+        log PaymentReceived(msg.sender, msg.sender, receivedAmount)
+    payedAmount: uint256 = 0
```

Comment:
Minor thing but my OCD obliges me to say that "payed" is only used in the nautical context and suggest to change this variable name to `paidAmount`

Reply:
aye aye 🏴‍☠️